### PR TITLE
chore(Java): add permutation sequences

### PR DESCRIPTION
**Pull request by Abhishek109062**
Added a new java file in Algorithms/Java/Maths/permutation_sequences.java

**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
- I have added new Algorithm
- Name - Permutation-Sequence
- Language - Java
- You can practice this algorithm online [Permutation-Sequence](https://leetcode.com/problems/permutation-sequence/)

**Other information:**

### CODE REVIEW

**Comment by Abhishek109062** [comment]
PLs review the new algorithm that i have added

File: algorithms/Java/Maths/permutation_sequence.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+Meaning: index to get number from is k / (n - 2)! = 1 / (4-2)! = 1 / 2! = 0.. from {1, 2, 4}, index 0 is 1
+
+
+so the numbers we have so far is 3, 1... and then repeating without explanations.
+
+
+{2, 4}
+
+k = k - (index from pervious) * (n-2)! = k - 0 * (n - 2)! = 1 - 0 = 1;
```
Comment (mohitchakraverty) [inline algorithms/Java/Maths/permutation_sequence.java:121]:
```suggestion
k = k - (index from previous) * (n-2)! = k - 0 * (n - 2)! = 1 - 0 = 1;
```

File: algorithms/Java/Maths/permutation_sequence.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+third number's index = k / (n - 3)! = 1 / (4-3)! = 1/ 1! = 1... from {2, 4}, index 1 has 4
+
+Third number is 4
+
+
+{2}
+
+k = k - (index from pervious) * (n - 3)! = k - 1 * (4 - 3)! = 1 - 1 = 0;
```
Comment (mohitchakraverty) [inline algorithms/Java/Maths/permutation_sequence.java:130]:
```suggestion
k = k - (index from previous) * (n - 3)! = k - 1 * (4 - 3)! = 1 - 1 = 0;
```

**Review by mohitchakraverty — CHANGES_REQUESTED** [review]
Please fix the typo and make the following changes

**Comment by mohitchakraverty** [comment]
Also update your branch **Abhishek109062:Permutation-Sequence** with **MakeContributions:main** to pass all the test cases

**Review by mohitchakraverty — APPROVED** [review]
Great work !

**Pull request merged**